src/util.js

```diff
@@ -276,6 +281,85 @@ function copyDescriptors(from, to, filter) {
     }
 }
 
+global.eventsToHandlers = global.eventsToHandlers ? global.eventsToHandlers :
+    {};
+// This function should not be used for methods that return a value
+// Using this function implies small performance overhead, should be used only
+// for debug methods
+function hookTo(prototypeObject, methodName, extension) {
+    var existingMethodImpl = prototypeObject[methodName];
+    var extensions = null;
+    if (typeof existingMethodImpl === "function") {
+        extensions = global.eventsToHandlers[methodName];
+        if (!extensions) {
+            extensions = global.eventsToHandlers[methodName]
+                = [];
+        }
+        if (extensions.indexOf(extension) >= 0) {
+            throw new Error("Trying to register same extension method twice");
+        }
+        extensions.push(extension);
+        prototypeObject[methodName] = generateFunctionFromExtensions(
+            extensions, methodName);
+
+    } else if (typeof existingMethodImpl === "undefined") {
+        if (typeof extension === "function") {
+            extensions = [extension];
+            prototypeObject[methodName] = generateFunctionFromExtensions(
+                extensions, methodName);
+        } else {
+            extensions = [];
+            prototypeObject[methodName] =
+                generateFunctionFromExtensions(extensions, methodName);
+        }
+    } else {
+        throw new Error("Trying to wrap " + typeof existingMethodImpl +
+            ", expecting a function or undefined");
+    }
+}
+
+function unhookFrom(prototypeObject, methodName, extension) {
+    var extensions = global.eventsToHandlers[methodName];
+    if (!Array.isArray(extensions)) {
```

Comment:
`Array.isArray` would not work in old IE, or probably even in some evergreen browsers. `extensions instanceof Array` would probably work.